skimage/measure/_marching_cubes_lewiner.py

```diff
@@ -1,10 +1,139 @@
+    normals : (V, 3) array
+        The normal direction at each vertex, as calculated from the
+        data.
+    values : (V, ) array
+        Gives a measure for the maximum value of the data in the local region
+        near each vertex. This can be used by visualization tools to apply
+        a colormap to the mesh.
+
+    Notes
+    -----
+    The algorithm [1] is an improved version of Chernyaev's Marching
+    Cubes 33 algorithm. It is an efficient algorithm that relies on
+    heavy use of lookup tables to handle the many different cases,
+    keeping the algorithm relatively easy. This implementation is
+    written in Cython, ported from Lewiner's C++ implementation.
+
+    To quantify the area of an isosurface generated by this algorithm, pass
+    verts and faces to `skimage.measure.mesh_surface_area`.
+
+    Regarding visualization of algorithm output, to contour a volume
+    named `myvolume` about the level 0.0, using the ``mayavi`` package::
+
+      >>>
+      >> from mayavi import mlab
+      >> verts, faces, _, _ = marching_cubes(myvolume, 0.0)
+      >> mlab.triangular_mesh([vert[0] for vert in verts],
+                              [vert[1] for vert in verts],
+                              [vert[2] for vert in verts],
+                              faces)
+      >> mlab.show()
+
+    Similarly using the ``visvis`` package::
+
+      >>>
+      >> import visvis as vv
+      >> verts, faces, normals, values = marching_cubes(myvolume, 0.0)
+      >> vv.mesh(np.fliplr(verts), faces, normals, values)
+      >> vv.use().Run()
+
+    References
+    ----------
+    .. [1] Thomas Lewiner, Helio Lopes, Antonio Wilson Vieira and Geovan
+           Tavares. Efficient implementation of Marching Cubes' cases with
+           topological guarantees. Journal of Graphics Tools 8(2)
+           pp. 1-15 (december 2003).
+           :DOI:`10.1080/10867651.2003.10487582`
+    .. [2] Lorensen, William and Harvey E. Cline. Marching Cubes: A High
+           Resolution 3D Surface Construction Algorithm. Computer Graphics
+           (SIGGRAPH 87 Proceedings) 21(4) July 1987, p. 163-170).
+           :DOI:`10.1145/37401.37422`
+
+    See Also
+    --------
+    skimage.measure.mesh_surface_area
+
+    """
+
+    use_classic = int(use_classic)
+
+    if use_classic == 0:
+        return _marching_cubes_lewiner(volume, level, spacing,
+                                       gradient_direction, step_size,
+                                       allow_degenerate, use_classic=False)
+    elif use_classic == 1:
+        return _marching_cubes_lewiner(volume, level, spacing,
+                                       gradient_direction, step_size,
+                                       allow_degenerate, use_classic=True)
+    elif use_classic == 2:
+        return _marching_cubes_classic(volume, level, spacing,
+                                       gradient_direction)
+    else:
+        raise ValueError("use_classic should be 0, 1 or 2.")
```

Comment:
I suggest `method=`, with some string or enum choices, and deprecating `use_classic`. Using an Int flag is very unpythonic.

Reply:
What about `method=` one of `lewiner`, `lorensen` and `_lorensen`?